packages/server/src/integrations/rest.ts

```diff
@@ -129,42 +130,82 @@ class RestIntegration implements IntegrationBase {
   }
 
   async parseResponse(response: any, pagination: PaginationConfig | null) {
-    let data, raw, headers
+    let data, raw, headers, presignedUrl, fileExtension, filename
+
     const contentType = response.headers.get("content-type") || ""
+    const contentDisposition = response.headers.get("content-disposition") || ""
+    const matches =
+      /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/.exec(contentDisposition) || []
+    filename = matches[1]?.replace(/['"]/g, "") || ""
+
     try {
-      if (response.status === 204) {
-        data = []
-        raw = []
-      } else if (contentType.includes("application/json")) {
-        data = await response.json()
-        raw = JSON.stringify(data)
-      } else if (
-        contentType.includes("text/xml") ||
-        contentType.includes("application/xml")
-      ) {
-        const rawXml = await response.text()
-        data =
-          (await xmlParser(rawXml, {
-            explicitArray: false,
-            trim: true,
-            explicitRoot: false,
-          })) || {}
-        // there is only one structure, its an array, return the array so it appears as rows
-        const keys = Object.keys(data)
-        if (keys.length === 1 && Array.isArray(data[keys[0]])) {
-          data = data[keys[0]]
+      if (filename) {
```

Comment:
This function is quite long and hard to read. Can we create subfunctions to make it more verbose?

Reply:
I split out both the file and xml handling part of this function into utils.